tests/test_lab_evaluate.py

```diff
@@ -0,0 +1,329 @@
+# Standard
+from unittest import mock
+from unittest.mock import patch
+
+# Third Party
+from click.testing import CliRunner
+
+# First Party
+from instructlab import lab
+
+
+def gen_qa_pairs(odd):
+    i = 1
+    qa_pairs = []
+    score = 0
+    while i < 5:
+        if i % 2:
+            if odd:
+                score = 0.2
+            else:
+                score = 0.1
+        elif not i % 2:
+            if odd:
+                score = 0.3
+            else:
+                score = 0.4
+        qa_pairs.append(
+            {
+                "question_id": i,
+                "score": score,
+                "qna_file": f"category{i}/qna.yaml",
+            }
+        )
+        i = i + 1
+    qa_pairs.append(
+        {
+            "question_id": i,
+            "score": 0.5,
+            "qna_file": f"category{i}/qna.yaml",
+        }
+    )
+    if odd:
+        qa_pairs.append(
+            {
+                "question_id": i + 1,
+                "score": 0.6,
+                "qna_file": f"category{i+1}/qna.yaml",
+            }
+        )
+    return qa_pairs
+
+
+def gen_individual_scores(odd):
+    i = 1
+    individual_scores = {}
+    score = 0
+    while i < 5:
+        if i % 2:
+            if odd:
+                score = 0.2
+            else:
+                score = 0.1
+        elif not i % 2:
+            if odd:
+                score = 0.3
+            else:
+                score = 0.4
+        individual_scores[f"task{i}"] = {"score": score}
+        i = i + 1
+    individual_scores[f"task{i}"] = {"score": 0.5}
+    return individual_scores
+
+
+class TestEvaluate:
+    @patch(
+        "instructlab.model.evaluate.launch_server",
+        return_value=(mock.MagicMock(), "http://120.0.0.1:8000/v1"),
```

Comment:
```suggestion
        return_value=(mock.MagicMock(), "http://127.0.0.1:8000/v1"),
```

I know it's a mock but shouldn't this be `127` for clarity? Or am I missing something?

Reply:
Typo, will fix.